instrumentation/opentelemetry_req/test/opentelemetry_req_test.exs

```diff
@@ -1,4 +1,120 @@
 defmodule OpentelemetryReqTest do
-  use ExUnit.Case
+  use ExUnit.Case, async: true
+
   doctest OpentelemetryReq
+
+  require Record
+
+  for {name, spec} <- Record.extract_all(from_lib: "opentelemetry/include/otel_span.hrl") do
+    Record.defrecord(name, spec)
+  end
+
+  for {name, spec} <- Record.extract_all(from_lib: "opentelemetry_api/include/opentelemetry.hrl") do
+    Record.defrecord(name, spec)
+  end
+
+  setup do
+    :otel_simple_processor.set_exporter(:otel_exporter_pid, self())
+
+    :ok
+  end
+
+  test "span name and attributes" do
+    {:ok, _resp} =
+      Req.new(adapter: fn req -> {req, Req.Response.new()} end)
+      |> OpentelemetryReq.attach()
+      |> Req.get(url: "http://example.com/users", span_name: "GET")
+
+    assert_receive {:span, span(name: "GET", kind: :client, attributes: attributes)}
+
+    assert %{
+             "http.url": "http://example.com/users",
+             "http.method": :GET,
+             "http.scheme": "http",
```

Comment:
This should be an atom.

Reply:
That's just what `URI.parse/1` returns, and which is the same behavior of Tesla middleware. Should we still change it?

I thought I have is, in case we still prefer atoms (perf reasons, right?), I can map to few known schemes and fallback to return as string. Does that seems a good compromise?

Reply:
It can just be to_atom but we can do that in another PR if it's a big lift. The [list of registered schemes](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml) is massive.